README
liveusb-creator
===============

This tool installs a Fedora LiveCD ISO on to a USB stick.

Requirements
============

  Linux
  -----
  o p7zip
  o syslinux

  Windows
  -------
  o If you wish to compile an exe, you will need to install py2exe.
    Alternatively, you can run the pre-compiled binary without any dependencies.
    http://lmacken.fedorapeople.org/liveusb-creator/liveusb-creator-1.0.zip

Using
=====

    See the wiki for instructions on how to use the liveusb-creator:

        https://fedorahosted.org/liveusb-creator

================================================================================

This tool is distributed with the following open source software:

   7-Zip
   http://www.7-zip.org
   Copyright (C) 1999-2007 Igor Pavlov.
   7-Zip is free software distributed under the GNU LGPL 
   (except for unRar code and AES code).

   SYSLINUX
   http://syslinux.zytor.com/
   Copyright 1994-2008 H. Peter Anvin - All Rights Reserved
   This program is free software; you can redistribute it and/or modify
   it under the terms of the GNU General Public License as published by
   the Free Software Foundation, Inc., 53 Temple Place Ste 330,
   Boston MA 02111-1307, USA; either version 2 of the License, or
   (at your option) any later version; incorporated herein by reference.

   dd for Windows
   http://www.chrysocome.net/dd
   dd is owned and copyright by Chrysocome and John Newbigin.
   It is made available under the terms of the GPLv2


ChangeLog
commit 0225dc48eb0e8f462b3082cd2aecb271decf7988
Author: Luke Macken <lmacken@redhat.com>
Date:   Wed Jun 24 14:09:58 2009 -0400

    v3.7

 liveusb-creator.spec |    5 ++++-
 setup.py             |    4 ++--
 2 files changed, 6 insertions(+), 3 deletions(-)

commit 71d73e76411e9b6d27db95c0dca567425e988c3a
Author: Luke Macken <lmacken@redhat.com>
Date:   Wed Jun 24 14:08:35 2009 -0400

    Hack around some odd Windows file permission issues

 liveusb/creator.py |    5 +++++
 1 files changed, 5 insertions(+), 0 deletions(-)

commit 52975b16377e1e9e6c9bf3041d2502a3044e6df7
Author: Luke Macken <lmacken@redhat.com>
Date:   Tue Jun 23 14:47:07 2009 -0400

    A little bit of unnecessary code removal

 liveusb/creator.py |   11 -----------
 liveusb/gui.py     |    1 -
 2 files changed, 0 insertions(+), 12 deletions(-)

commit 7377b2a861c9737a67b77dca5bf15095ab68eaf9
Author: Luke Macken <lmacken@redhat.com>
Date:   Tue Jun 23 14:38:52 2009 -0400

    Drop a warning if the MBR doesn't match syslinux, and don't force people to reset.

 liveusb/creator.py |    4 ++--
 liveusb/gui.py     |   14 +++++++++-----
 2 files changed, 11 insertions(+), 7 deletions(-)

commit 2ef0e7cf4f57a9e3b9b7cbde53e76579d7ce38b3
Author: Luke Macken <lmacken@redhat.com>
Date:   Tue Jun 23 14:38:12 2009 -0400

    Update Sugar on a Stick to the new "Strawberry" release

 liveusb/releases.py |    4 ++--
 1 files changed, 2 insertions(+), 2 deletions(-)

commit 0c77c9883bd6a3dbc0848e5a3c4fe078ec85ad06
Author: Luke Macken <lmacken@redhat.com>
Date:   Fri Jun 19 11:51:54 2009 -0400

    Refactor some of our mbr checking code, so it is essentially a NOOP in Windows

 liveusb/creator.py |   93 ++++++++++++++++++++++++++++++----------------------
 1 files changed, 54 insertions(+), 39 deletions(-)

commit 27b28fad946b44b958f1d2c00a215e621ef5e0d6
Author: Luke Macken <lmacken@redhat.com>
Date:   Fri Jun 19 11:48:25 2009 -0400

    Make our MBR checking work on devices w/o parents

 liveusb/creator.py |    4 ++--
 1 files changed, 2 insertions(+), 2 deletions(-)

commit 37805207d925c1abaed69dc6e78045f2df36fe75
Author: Luke Macken <lmacken@redhat.com>
Date:   Fri Jun 19 02:40:14 2009 -0400

    Don't clear the text area when another stick is inserted

 liveusb/gui.py |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

commit d77bb67eceb97f6d85430936ae04df563c929b25
Author: Luke Macken <lmacken@redhat.com>
Date:   Fri Jun 19 02:39:53 2009 -0400

    Allow people to pass the ISO as a command line argument

 liveusb/gui.py |   18 +++++++++++++-----
 1 files changed, 13 insertions(+), 5 deletions(-)

commit 537cff6bf344bad6308e88fd466ace3bf5a5a0be
Author: Luke Macken <lmacken@redhat.com>
Date:   Fri Jun 19 02:38:54 2009 -0400

    Don't try and unmount th device every time the window is closed

 liveusb/creator.py |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

commit cc462b30c4fdde854a6d09984fccc2ccd0377453
Author: Luke Macken <lmacken@redhat.com>
Date:   Fri Jun 19 02:38:42 2009 -0400

    Some minor command formatting tweaks

 liveusb/creator.py |   10 ++++++----
 1 files changed, 6 insertions(+), 4 deletions(-)

commit dcf9a57f7a8a0069d061034a0b9fdcb3d76b3107
Author: Luke Macken <lmacken@redhat.com>
Date:   Fri Jun 19 02:38:22 2009 -0400

    Fix our ldlinux.sys cleanup code

 liveusb/creator.py |    3 ++-
 1 files changed, 2 insertions(+), 1 deletions(-)

commit e6ddd4f673350615a108ecde825789753766e15a
Author: Luke Macken <lmacken@redhat.com>
Date:   Fri Jun 19 02:37:59 2009 -0400

    Make our unmount method a bit more robust.
    
    If HAL fails, call unmount directly...

 liveusb/creator.py |   29 +++++++++++++++++++----------
 1 files changed, 19 insertions(+), 10 deletions(-)

commit 7c5d3392ae93d7f98702df128ee26cb79edd86b8
Author: Luke Macken <lmacken@redhat.com>
Date:   Fri Jun 19 02:37:45 2009 -0400

    Bump version for next release

 liveusb-creator |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

commit 5084936157d3f1115f239f06bab32aeaa713aa4c
Author: Luke Macken <lmacken@redhat.com>
Date:   Fri Jun 19 02:36:03 2009 -0400

    Ensure the MBR matches the installed syslinux mbr.bin.
    
    When zeroing out your drive, and then laying down a fresh msdos partition table and FAT32 partition in Gparted, it seems to set an invalid MBR (b8fa).
    This patch will ensure that your MBR matches the system-wide mbr.bin before
    continuing.  (#506886)

 liveusb-creator    |    3 +++
 liveusb/creator.py |   47 +++++++++++++++++++++++++++++++++++++----------
 liveusb/gui.py     |   17 +++++++++++++++++
 3 files changed, 57 insertions(+), 10 deletions(-)

commit 621ca1694fdfa6c73fcbfd3d35d8dc05fd605228
Author: Luke Macken <lmacken@redhat.com>
Date:   Thu Jun 18 12:30:56 2009 -0400

    Update a unit test to handle sha256 checksums in our liveusb.releases file

 tests/test_creator.py |    4 +++-
 1 files changed, 3 insertions(+), 1 deletions(-)

commit 48f15fa50ca7da1c928b086c465596a0b52ae1df
Author: raven <raven@fedoraproject.org>
Date:   Sun Jun 14 18:14:18 2009 +0000

    Sending translation for Polish

 po/pl.po |   38 +++++++++++++++++++++++++-------------
 1 files changed, 25 insertions(+), 13 deletions(-)

commit 1000313a7638c4e13a3c3c01f682b53b8fc498cd
Author: Luke Macken <lmacken@redhat.com>
Date:   Fri Jun 12 16:37:49 2009 -0400

    v3.6.8

 liveusb-creator.spec |    5 ++++-
 setup.py             |    4 ++--
 2 files changed, 6 insertions(+), 3 deletions(-)

commit 1415f61541477a47623797ec0239de1038fcfbfa
Author: Luke Macken <lmacken@redhat.com>
Date:   Fri Jun 12 16:33:23 2009 -0400

    Remove some stray print statements

 liveusb/creator.py |    2 --
 1 files changed, 0 insertions(+), 2 deletions(-)

commit 5487a20c6902fd0124dc503055497be7ea5509ae
Author: Luke Macken <lmacken@redhat.com>
Date:   Fri Jun 12 16:33:12 2009 -0400

    Don't prompt about overwriting files from mtools (#491234)

 liveusb/creator.py |   23 +++++++++++++++++------

liveusb/dialog.py
# -*- coding: utf-8 -*-

# Form implementation generated from reading ui file 'data/liveusb-creator.ui'
#
# Created: Thu May 29 22:56:03 2008
#      by: PyQt4 UI code generator 4.3.3
#
# WARNING! All changes made in this file will be lost!

from PyQt4 import QtCore, QtGui
from liveusb import _

class Ui_Dialog(object):
    def setupUi(self, Dialog):
        Dialog.setObjectName("Dialog")
        Dialog.resize(QtCore.QSize(QtCore.QRect(0,0,422,388).size()).expandedTo(Dialog.minimumSizeHint()))

        sizePolicy = QtGui.QSizePolicy(QtGui.QSizePolicy.Fixed,QtGui.QSizePolicy.Fixed)
        sizePolicy.setHorizontalStretch(0)
        sizePolicy.setVerticalStretch(0)
        sizePolicy.setHeightForWidth(Dialog.sizePolicy().hasHeightForWidth())
        Dialog.setSizePolicy(sizePolicy)

        self.startButton = QtGui.QPushButton(Dialog)
        self.startButton.setEnabled(True)
        self.startButton.setGeometry(QtCore.QRect(130,350,158,34))
        self.startButton.setObjectName("startButton")

        self.textEdit = QtGui.QTextEdit(Dialog)
        self.textEdit.setGeometry(QtCore.QRect(10,200,401,111))

        font = QtGui.QFont()
        font.setPointSize(8)
        self.textEdit.setFont(font)
        self.textEdit.setReadOnly(True)
        self.textEdit.setObjectName("textEdit")

        self.progressBar = QtGui.QProgressBar(Dialog)
        self.progressBar.setGeometry(QtCore.QRect(10,320,401,23))
        self.progressBar.setProperty("value",QtCore.QVariant(0))
        self.progressBar.setObjectName("progressBar")

        self.downloadGroup = QtGui.QGroupBox(Dialog)
        self.downloadGroup.setGeometry(QtCore.QRect(210,80,201,51))

        font = QtGui.QFont()
        font.setPointSize(8)
        self.downloadGroup.setFont(font)
        self.downloadGroup.setObjectName("downloadGroup")

        self.downloadCombo = QtGui.QComboBox(self.downloadGroup)
        self.downloadCombo.setGeometry(QtCore.QRect(10,20,181,22))
        self.downloadCombo.setObjectName("downloadCombo")

        self.label_2 = QtGui.QLabel(Dialog)
        self.label_2.setGeometry(QtCore.QRect(180,100,23,24))

        font = QtGui.QFont()
        font.setFamily("Verdana")
        font.setPointSize(10)
        font.setWeight(75)
        font.setBold(True)
        self.label_2.setFont(font)
        self.label_2.setObjectName("label_2")

        self.groupBox = QtGui.QGroupBox(Dialog)
        self.groupBox.setGeometry(QtCore.QRect(10,80,161,51))

        font = QtGui.QFont()
        font.setPointSize(8)
        self.groupBox.setFont(font)
        self.groupBox.setObjectName("groupBox")

        self.isoBttn = QtGui.QPushButton(self.groupBox)
        self.isoBttn.setGeometry(QtCore.QRect(11,18,141,25))
        self.isoBttn.setObjectName("isoBttn")

        self.groupBox_2 = QtGui.QGroupBox(Dialog)
        self.groupBox_2.setGeometry(QtCore.QRect(10,140,191,51))

        font = QtGui.QFont()
        font.setPointSize(8)
        self.groupBox_2.setFont(font)
        self.groupBox_2.setObjectName("groupBox_2")

        self.driveBox = QtGui.QComboBox(self.groupBox_2)
        self.driveBox.setGeometry(QtCore.QRect(10,20,145,20))
        self.driveBox.setEditable(False)
        self.driveBox.setInsertPolicy(QtGui.QComboBox.InsertAtTop)
        self.driveBox.setDuplicatesEnabled(False)
        self.driveBox.setObjectName("driveBox")

        self.refreshDevicesButton = QtGui.QPushButton(self.groupBox_2)
        self.refreshDevicesButton.setGeometry(QtCore.QRect(156,16,30,26))
        self.refreshDevicesButton.setIcon(QtGui.QIcon(":/refresh.png"))
        self.refreshDevicesButton.setFlat(True)
        self.refreshDevicesButton.setObjectName("refreshDevicesButton")

        self.overlayTitle = QtGui.QGroupBox(Dialog)
        self.overlayTitle.setGeometry(QtCore.QRect(210,140,201,51))

        font = QtGui.QFont()
        font.setPointSize(8)
        self.overlayTitle.setFont(font)
        self.overlayTitle.setObjectName("overlayTitle")

        self.overlaySlider = QtGui.QSlider(self.overlayTitle)
        self.overlaySlider.setGeometry(QtCore.QRect(10,20,181,21))
        self.overlaySlider.setMaximum(2047)
        self.overlaySlider.setOrientation(QtCore.Qt.Horizontal)
        self.overlaySlider.setTickPosition(QtGui.QSlider.NoTicks)
        self.overlaySlider.setObjectName("overlaySlider")

        self.label = QtGui.QLabel(Dialog)
        self.label.setGeometry(QtCore.QRect(0,0,430,72))
        self.label.setPixmap(QtGui.QPixmap(":/liveusb-header.png"))
        self.label.setObjectName("label")

        self.retranslateUi(Dialog)
        QtCore.QMetaObject.connectSlotsByName(Dialog)

    def retranslateUi(self, Dialog):
        Dialog.setWindowTitle(QtGui.QApplication.translate("Dialog", _("Fedora LiveUSB Creator"), None, QtGui.QApplication.UnicodeUTF8))
        self.startButton.setWhatsThis(QtGui.QApplication.translate("Dialog", _("This button will begin the LiveUSB creation process.  This entails optionally downloading a release (if an existing one wasn\'t selected),  extracting the ISO to the USB device, creating the persistent overlay, and installing the bootloader."), None, QtGui.QApplication.UnicodeUTF8))
        self.startButton.setText(QtGui.QApplication.translate("Dialog", _("Create Live USB"), None, QtGui.QApplication.UnicodeUTF8))
        self.textEdit.setWhatsThis(QtGui.QApplication.translate("Dialog", _("This is the status console, where all messages get written to."), None, QtGui.QApplication.UnicodeUTF8))
        self.progressBar.setWhatsThis(QtGui.QApplication.translate("Dialog", _("This is the progress bar that will indicate how far along in the LiveUSB creation process you are"), None, QtGui.QApplication.UnicodeUTF8))
        self.downloadGroup.setWhatsThis(QtGui.QApplication.translate("Dialog", _("If you do not select an existing Live CD, the selected release will be downloaded for you."), None, QtGui.QApplication.UnicodeUTF8))
        self.downloadGroup.setTitle(QtGui.QApplication.translate("Dialog", _("Download Fedora"), None, QtGui.QApplication.UnicodeUTF8))
        self.label_2.setText(QtGui.QApplication.translate("Dialog", _("or"), None, QtGui.QApplication.UnicodeUTF8))
        self.groupBox.setWhatsThis(QtGui.QApplication.translate("Dialog", _("This button allows you to browse for an existing Live CD ISO that you have previously downloaded.  If you do not select one, a release will be downloaded for you automatically."), None, QtGui.QApplication.UnicodeUTF8))
        self.groupBox.setTitle(QtGui.QApplication.translate("Dialog", _("Use existing Live CD"), None, QtGui.QApplication.UnicodeUTF8))
        self.isoBttn.setText(QtGui.QApplication.translate("Dialog", _("Browse"), None, QtGui.QApplication.UnicodeUTF8))
        self.isoBttn.setShortcut(QtGui.QApplication.translate("Dialog", "Alt+B", None, QtGui.QApplication.UnicodeUTF8))
        self.groupBox_2.setWhatsThis(QtGui.QApplication.translate("Dialog", _("This is the USB stick that you want to install your Live CD on.  This device must be formatted with the FAT filesystem."), None, QtGui.QApplication.UnicodeUTF8))
        self.groupBox_2.setTitle(QtGui.QApplication.translate("Dialog", _("Target Device"), None, QtGui.QApplication.UnicodeUTF8))
        self.overlayTitle.setWhatsThis(QtGui.QApplication.translate("Dialog", _("By allocating extra space on your USB stick for a persistent overlay, you will be able to store data and make permanent modifications to your live operating system.  Without it, you will not be able to save data that will persist after a reboot."), None, QtGui.QApplication.UnicodeUTF8))
        self.overlayTitle.setTitle(QtGui.QApplication.translate("Dialog", _("Persistent Storage (0 Mb)"), None, QtGui.QApplication.UnicodeUTF8))

import resources_rc
